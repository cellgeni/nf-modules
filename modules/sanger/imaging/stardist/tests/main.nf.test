// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test imaging/stardist
nextflow_process {

    name "Test Process IMAGING_STARDIST"
    script "../main.nf"
    process "IMAGING_STARDIST"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/stardist"

    test("stardist - tif") {

        when {
            process {
                """

                input[0] = [
                    [ id:'test' ], // meta map
                    0,  0, 500, 500, // x_min, x_max, y_min, y_max
                    file(params.modules_testdata_base_path + 'imaging/segmentation/nuclear_image.tif', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("stardist - tif - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    0,  0, 500, 500, // x_min, x_max, y_min, y_max
                    file(params.modules_testdata_base_path + 'imaging/segmentation/nuclear_image.tif', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
