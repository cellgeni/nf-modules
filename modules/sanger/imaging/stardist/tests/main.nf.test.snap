{
    "stardist - tif - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "versions": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-18T22:13:45.973377162"
    },
    "stardist - tif": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,110576d3df4397edcf0ee3f2ad097a9b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "versions": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,110576d3df4397edcf0ee3f2ad097a9b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-18T22:13:35.658406973"
    }
}